server/neptune/workflows/internal/terraform/state/store.go

```diff
@@ -85,6 +87,18 @@ func (s *WorkflowStore) UpdatePlanJobWithStatus(status JobStatus, options ...Upd
 }
 
 func (s *WorkflowStore) UpdateApplyJobWithStatus(status JobStatus, options ...UpdateOptions) error {
+	var timeStamp time.Time
+	for _, o := range options {
+		timeStamp = o.Timestamp
```

Comment:
If there are multiple options and one is nil, this would get overwritten?

Reply:
Ahh right, I have created helper functions `getStartTime` and `getEndTime` that finds the first non-empty start time and end time.